qa: Intermittent failure in `wallet_encryption.py --descriptors`

https://cirrus-ci.com/task/4615142003441664

<details><summary>log excerpt</summary>
<p>

```
177/244 - wallet_encryption.py --descriptors failed, Duration: 6 s

stdout:
2022-06-27T07:50:03.095000Z TestFramework (INFO): Initializing test directory C:\Users\ContainerAdministrator\AppData\Local\Temp\test_runner_₿_🏃_20220627_071202\wallet_encryption_65

2022-06-27T07:50:08.820000Z TestFramework (INFO): Check a timeout less than the limit

2022-06-27T07:50:09.007000Z TestFramework (ERROR): Assertion failed

Traceback (most recent call last):

  File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\test_framework.py", line 133, in main

    self.run_test()

  File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\wallet_encryption.py", line 85, in run_test

    assert_greater_than(expected_time_with_buffer, actual_time)

  File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\util.py", line 56, in assert_greater_than

    raise AssertionError("%s <= %s" % (str(thing1), str(thing2)))

AssertionError: 1756315608.992383 <= 1756315609

2022-06-27T07:50:09.070000Z TestFramework (INFO): Stopping nodes

2022-06-27T07:50:09.211000Z TestFramework (WARNING): Not cleaning up dir C:\Users\ContainerAdministrator\AppData\Local\Temp\test_runner_₿_🏃_20220627_071202\wallet_encryption_65

2022-06-27T07:50:09.211000Z TestFramework (ERROR): Test failed. Test logging available at C:\Users\ContainerAdministrator\AppData\Local\Temp\test_runner_₿_🏃_20220627_071202\wallet_encryption_65/test_framework.log

2022-06-27T07:50:09.211000Z TestFramework (ERROR): 

2022-06-27T07:50:09.211000Z TestFramework (ERROR): Hint: Call C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\combine_logs.py 'C:\Users\ContainerAdministrator\AppData\Local\Temp\test_runner_₿_🏃_20220627_071202\wallet_encryption_65' to consolidate all logs

2022-06-27T07:50:09.211000Z TestFramework (ERROR): 

2022-06-27T07:50:09.211000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.

2022-06-27T07:50:09.211000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues

2022-06-27T07:50:09.211000Z TestFramework (ERROR): 
```

</p>
</details>

I have the same failing at Mac OS 

```
test/functional/wallet_descriptor.py --descriptors
2022-07-07T04:25:35.600000Z TestFramework (INFO): Initializing test directory /var/folders/fy/kkvn4xds2cg0zgd2ckkhntsm0000gn/T/bitcoin_func_test_xhh_y_g9
2022-07-07T04:25:40.993000Z TestFramework (WARNING): Skipping BDB test
2022-07-07T04:25:40.993000Z TestFramework (INFO): Making a descriptor wallet
2022-07-07T04:25:41.047000Z TestFramework (INFO): Checking wallet info
2022-07-07T04:25:41.047000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/Users/saranshsharma/Documents/bitcoin/test/functional/test_framework/test_framework.py", line 133, in main
    self.run_test()
  File "/Users/saranshsharma/Documents/bitcoin/test/functional/wallet_descriptor.py", line 44, in run_test
    assert_equal(wallet_info['keypoolsize'], 400)
  File "/Users/saranshsharma/Documents/bitcoin/test/functional/test_framework/util.py", line 51, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(300 == 400)
2022-07-07T04:25:41.104000Z TestFramework (INFO): Stopping nodes
2022-07-07T04:25:41.322000Z TestFramework (WARNING): Not cleaning up dir /var/folders/fy/kkvn4xds2cg0zgd2ckkhntsm0000gn/T/bitcoin_func_test_xhh_y_g9
2022-07-07T04:25:41.323000Z TestFramework (ERROR): Test failed. Test logging available at /var/folders/fy/kkvn4xds2cg0zgd2ckkhntsm0000gn/T/bitcoin_func_test_xhh_y_g9/test_framework.log
2022-07-07T04:25:41.323000Z TestFramework (ERROR): 
2022-07-07T04:25:41.323000Z TestFramework (ERROR): Hint: Call /Users/saranshsharma/Documents/bitcoin/test/functional/combine_logs.py '/var/folders/fy/kkvn4xds2cg0zgd2ckkhntsm0000gn/T/bitcoin_func_test_xhh_y_g9' to consolidate all logs
2022-07-07T04:25:41.323000Z TestFramework (ERROR): 
2022-07-07T04:25:41.323000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2022-07-07T04:25:41.323000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
2022-07-07T04:25:41.324000Z TestFramework (ERROR): 

```

> I tried logging the Wallet info the following prints
> 
> ```
> {'walletname': 'desc1', 'walletversion': 169900, 'format': 'sqlite', 'balance': Decimal('0E-8'), 'unconfirmed_balance': Decimal('0E-8'), 'immature_balance': Decimal('0E-8'), 'txcount': 0, 'keypoolsize': 300, 'keypoolsize_hd_internal': 300, 'paytxfee': Decimal('0E-8'), 'private_keys_enabled': True, 'avoid_reuse': False, 'scanning': False, 'descriptors': True}
> ```

The keypoolsize is set for 400 , and  keypoolsize_hd_internal and the test is asserting for 400 could be an init issues when generating wallet. (Don't know maybe)

But interestingly it fails after manually setting 300 in the assertion to check if it works 

```
test/functional/wallet_descriptor.py --descriptors
2022-07-07T04:32:17.026000Z TestFramework (INFO): Initializing test directory /var/folders/fy/kkvn4xds2cg0zgd2ckkhntsm0000gn/T/bitcoin_func_test__r2j28ed
2022-07-07T04:32:17.576000Z TestFramework (WARNING): Skipping BDB test
2022-07-07T04:32:17.576000Z TestFramework (INFO): Making a descriptor wallet
2022-07-07T04:32:17.632000Z TestFramework (INFO): Checking wallet info
2022-07-07T04:32:17.632000Z TestFramework (INFO): {'walletname': 'desc1', 'walletversion': 169900, 'format': 'sqlite', 'balance': Decimal('0E-8'), 'unconfirmed_balance': Decimal('0E-8'), 'immature_balance': Decimal('0E-8'), 'txcount': 0, 'keypoolsize': 300, 'keypoolsize_hd_internal': 300, 'paytxfee': Decimal('0E-8'), 'private_keys_enabled': True, 'avoid_reuse': False, 'scanning': False, 'descriptors': True}
2022-07-07T04:32:17.632000Z TestFramework (INFO): Test that getnewaddress and getrawchangeaddress work
2022-07-07T04:32:17.703000Z TestFramework (INFO): Test sending and receiving
2022-07-07T04:32:17.705000Z TestFramework (INFO): Test disabled RPCs
2022-07-07T04:32:17.706000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/Users/saranshsharma/Documents/bitcoin/test/functional/test_framework/util.py", line 134, in try_rpc
    fun(*args, **kwds)
  File "/Users/saranshsharma/Documents/bitcoin/test/functional/test_framework/coverage.py", line 49, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/Users/saranshsharma/Documents/bitcoin/test/functional/test_framework/authproxy.py", line 144, in __call__
    raise JSONRPCException(response['error'], status)
test_framework.authproxy.JSONRPCException: This type of wallet does not support this command (-4)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/saranshsharma/Documents/bitcoin/test/functional/test_framework/test_framework.py", line 133, in main
    self.run_test()
  File "/Users/saranshsharma/Documents/bitcoin/test/functional/wallet_descriptor.py", line 97, in run_test
    assert_raises_rpc_error(-4, "Only legacy wallets are supported by this command", recv_wrpc.rpc.importprivkey, "cVpF924EspNh8KjYsfhgY96mmxvT6DgdWiTYMtMjuM74hJaU5psW")
  File "/Users/saranshsharma/Documents/bitcoin/test/functional/test_framework/util.py", line 125, in assert_raises_rpc_error
    assert try_rpc(code, message, fun, *args, **kwds), "No exception raised"
  File "/Users/saranshsharma/Documents/bitcoin/test/functional/test_framework/util.py", line 140, in try_rpc
    raise AssertionError(
AssertionError: Expected substring not found in error message:
substring: 'Only legacy wallets are supported by this command'
error message: 'This type of wallet does not support this command'.
2022-07-07T04:32:17.761000Z TestFramework (INFO): Stopping nodes
2022-07-07T04:32:17.979000Z TestFramework (WARNING): Not cleaning up dir /var/folders/fy/kkvn4xds2cg0zgd2ckkhntsm0000gn/T/bitcoin_func_test__r2j28ed
2022-07-07T04:32:17.979000Z TestFramework (ERROR): Test failed. Test logging available at /var/folders/fy/kkvn4xds2cg0zgd2ckkhntsm0000gn/T/bitcoin_func_test__r2j28ed/test_framework.log
2022-07-07T04:32:17.980000Z TestFramework (ERROR): 
2022-07-07T04:32:17.980000Z TestFramework (ERROR): Hint: Call /Users/saranshsharma/Documents/bitcoin/test/functional/combine_logs.py '/var/folders/fy/kkvn4xds2cg0zgd2ckkhntsm0000gn/T/bitcoin_func_test__r2j28ed' to consolidate all logs
2022-07-07T04:32:17.980000Z TestFramework (ERROR): 
2022-07-07T04:32:17.980000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2022-07-07T04:32:17.981000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
2022-07-07T04:32:17.981000Z TestFramework (ERROR): 

```

--- a/test/functional/wallet_encryption.py
+++ b/test/functional/wallet_encryption.py
@@ -9,8 +9,7 @@ import time
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import (
     assert_raises_rpc_error,
-    assert_greater_than,
-    assert_greater_than_or_equal,
+    assert_equal,
 )
 
 
@@ -76,21 +75,18 @@ class WalletEncryptionTest(BitcoinTestFramework):
 
         self.log.info('Check a timeout less than the limit')
         MAX_VALUE = 100000000
-        expected_time = int(time.time()) + MAX_VALUE - 600
+        now = int(time.time())
+        self.nodes[0].setmocktime(now)
+        expected_time = now + MAX_VALUE - 600
         self.nodes[0].walletpassphrase(passphrase2, MAX_VALUE - 600)
-        # give buffer for walletpassphrase, since it iterates over all encrypted keys
-        expected_time_with_buffer = time.time() + MAX_VALUE - 600
         actual_time = self.nodes[0].getwalletinfo()['unlocked_until']
-        assert_greater_than_or_equal(actual_time, expected_time)
-        assert_greater_than(expected_time_with_buffer, actual_time)
+        assert_equal(actual_time, expected_time)
 
         self.log.info('Check a timeout greater than the limit')
-        expected_time = int(time.time()) + MAX_VALUE - 1
+        expected_time = now + MAX_VALUE
         self.nodes[0].walletpassphrase(passphrase2, MAX_VALUE + 1000)
-        expected_time_with_buffer = time.time() + MAX_VALUE
         actual_time = self.nodes[0].getwalletinfo()['unlocked_until']
-        assert_greater_than_or_equal(actual_time, expected_time)
-        assert_greater_than(expected_time_with_buffer, actual_time)
+        assert_equal(actual_time, expected_time)
 
 
 if __name__ == '__main__':

